pandas/tests/series/test_analytics.py

```diff
@@ -1045,6 +1030,89 @@ def test_drop_duplicates_bool(self):
+    ])
+    def test_duplicated_keep(self, keep, expected):
+        s = Series(['a', 'b', 'b', 'c', 'a'], name='name')
+
+        result = s.duplicated(keep=keep)
+        tm.assert_series_equal(result, expected)
+
+    @pytest.mark.parametrize('keep, expected', [
+        ('first', Series([False, False, True, False, True])),
+        ('last', Series([True, True, False, False, False])),
+        (False, Series([True, True, True, False, True]))
+    ])
+    def test_duplicated_nan_none(self, keep, expected):
+        # np.nan and None are considered equal
+        s = Series([np.nan, 3, 3, None, np.nan], dtype=object)
+
+        result = s.duplicated(keep=keep)
+        tm.assert_series_equal(result, expected)
+
+    @pytest.mark.parametrize('keep, expected_inv_values', [
+        ('first', [1, 4, 4, 16, 1]),
+        ('last', [25, 9, 9, 16, 25])
+    ])
+    def test_duplicated_inverse(self, keep, expected_inv_values):
+        # check that return_inverse kwarg does not affect outcome;
+        # index of inverse must be correctly transformed as well
+        idx = [1, 4, 9, 16, 25]
+        s = Series(['a', 'b', 'b', 'c', 'a'], index=idx)
+
+        expected_isdup = s.duplicated(keep=keep)
+        expected_inv = Series(expected_inv_values, index=idx)
+        result_isdup, result_inv = s.duplicated(keep=keep,
+                                                return_inverse=True)
+        tm.assert_series_equal(result_isdup, expected_isdup)
+        tm.assert_series_equal(result_inv, expected_inv)
+
+        # test that inv works (and fits together with expected_isdup)
+        unique = s.loc[~expected_isdup]
+        reconstr = unique.reindex(result_inv)
+        # Series has no set_index (GH21684)
+        reconstr.index = result_inv.index
+        tm.assert_series_equal(reconstr, s)
+
+    def test_duplicated_inverse_raises(self):
+        s = Series(['a', 'b', 'b', 'c', 'a'])
+
+        # keep = False
+        rgx = 'The parameters return_inverse=True and keep=False cannot be.*'
+        with tm.assert_raises_regex(ValueError, rgx):
+            s.duplicated(keep=False, return_inverse=True)
+
+    @pytest.mark.parametrize('keep', ['first', 'last'])
+    def test_duplicated_inverse_large(self, keep):
+        # unsorted index important to check 'first'/'last' functionality
+        s = Series(np.random.randint(0, 1000, 10000)).sample(5000)
+
+        expected_isdup = s.duplicated(keep=keep)
+        result_isdup, result_inv = s.duplicated(keep=keep, return_inverse=True)
+        tm.assert_series_equal(result_isdup, expected_isdup)
+
+        # test that inv works (and fits together with expected_isdup)
+        unique = s.loc[~expected_isdup]
+        reconstr = unique.reindex(result_inv)
+        # Series has no set_index (GH21684)
+        reconstr.index = result_inv.index
+        tm.assert_series_equal(reconstr, s)
+
+    @pytest.mark.parametrize('keep', ['first', 'last'])
+    def test_duplicated_inverse_fastpath(self, keep):
```

Comment:
What is the point of all of the `fastpath` tests? Couldn't an array of all distinct values simply be sent via parametrization to the other tests?

Reply:
I just made a separate test to make sure that path was hit. I don't fully understand the parametrization request - each test should run on an array and its deduplicated version?